blockchain/blockchain.go

```diff
@@ -159,8 +159,7 @@ type BlockChain struct {
 	scope         event.SubscriptionScope
 	genesisBlock  *types.Block
 
-	mu      sync.RWMutex // global mutex for locking chain operations
-	chainmu sync.RWMutex // blockchain insertion lock
+	mu sync.RWMutex // global mutex for locking chain operations
```

Comment:
What about bringing https://github.com/ethereum/go-ethereum/pull/22853? This changes the mutex type to ClosableMutexType

Reply:
If it's needed, I'll bring it in another PR.
This PR is only about getting rid of `chainmu`.